jqm-all/jqm-ws/src/main/java/com/enioka/jqm/ws/api/ServiceSimpleStandalone.java

```diff
@@ -0,0 +1,146 @@
+@JakartarsResource
+@Path("/simple")
+public class ServiceSimpleStandalone extends ServiceSimple
+{
+    private final static Logger log = LoggerFactory.getLogger(ServiceSimpleStandalone.class);
+
+    private Integer jqmNodeId = null;
+
+    private String localIp;
+
+    @Activate
+    @Override
+    public void onServiceActivation(Map<String, Object> properties)
+    {
+        if (!properties.containsKey("jqmnodeid"))
+        {
+            throw new JqmAdminApiException("OSGi configuration for ServiceSimple does not contain a valid node ID");
+        }
+
+        jqmNodeId = (int) properties.get("jqmnodeid"); // small int - cannot be null.
+        log.info("\tStarting ServiceSimple with node ID {}", jqmNodeId);
+
+        try {
+            localIp = Inet4Address.getLocalHost().getHostAddress();
+            // TODO: reset ID sequence of the local engine using `idSequenceBaseFromIp(localIp)`
+        } catch (UnknownHostException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /////////////////////////////////////////////////////////////
+    // ID <> IP helpers
+    /////////////////////////////////////////////////////////////
+
+    /**
+     * Extracts the IP address corresponding to an ID.
+     * <p>
+     * Each ID, when divided by 1 million, results in the full IPv4 address (each part with padding 0's) of the node
+     * that created it.
+     *
+     * @param id The ID
+     * @return The corresponding IP address
+     */
+    public static String ipFromId(long id) {
+        final var base = id / 1_000_000;
+        return "" +
+            base / 1000L / 1000L / 1000 % 1000L +
+            '.' +
+            base / 1000L / 1000L % 1000L +
+            '.' +
+            base / 1000L % 1000L +
+            '.' +
+            base % 1000L;
+    }
+
+    /**
+     * Computes the base (first value) of an ID sequence for a given IP address.
+     * <p>
+     * The sequence's starting value is the concatenation of each 0-padded part of the IPv4,
+     * multiplied by 1 million (which gives us 1 million values before a collision with the next IP).
+     *
+     * @param ip The IP address
+     * @return The first value of the corresponding sequence
+     */
+    public static long idSequenceBaseFromIp(String ip) {
+        return Long.parseLong(Arrays.stream(ip.split("\\."))
+            .map(stringValue -> String.format("%03d", Integer.parseInt(stringValue)))
+            .collect(Collectors.joining())
+        ) * 1_000_000L;
+    }
+
+    /////////////////////////////////////////////////////////////
+    // The one and only really simple API
+    /////////////////////////////////////////////////////////////
+
+    @GET
+    @Path("status")
+    @Produces(MediaType.TEXT_PLAIN)
+    //@Override
+    public String getStatus(@QueryParam("id") int id)
+    {
+        final var realDestIp = ipFromId(id);
+        JqmClient client = null;
```

Comment:
may bet set the client in the constructor to avoid instanciation every time is call ?

Reply:
Not possible, the JerseyClient needs to be re-created every time because the IP it points to will be different. The factory should cache the JdbcClient by default so it should not be costly.